frontend/src/components/SeedlotRegistrationProgress/InterimStorage/index.tsx

```diff
@@ -0,0 +1,371 @@
+import { useAuth } from '../../../contexts/AuthContext';
+import ApplicantInfo from '../../../types/ApplicantInfo';
+import './styles.scss';
+
+interface InterimStorageStepProps {
+  setStep: Function
+}
+
+const InterimStorage = ({ setStep }: InterimStorageStepProps) => {
+  const { token } = useAuth();
+  const { seedlot } = useParams();
+
+  const mockAgencyOptions: Array<string> = [
+    '0032 - Strong Seeds Orchard - SSO',
+    '0035 - Weak Seeds Orchard - WSO',
+    '0038 - Okay Seeds Orchard - OSO'
+  ];
+
+  const mockStartDataMin: string = '03/20/2023';
+  const mockStartDataMax: string = '03/23/2023';
+  const mockEndDataMin: string = '03/20/2023';
+  const mockEndDataMax: string = '03/23/2023';
+
+  const getAxiosConfig = () => {
+    const axiosConfig = {};
+    if (token) {
+      const headers = {
+        headers: {
+          Authorization: `Bearer ${token}`
+        }
+      };
+      Object.assign(axiosConfig, headers);
+    }
+    return axiosConfig;
+  };
+
+  const nameInputRef = useRef<HTMLInputElement>(null);
+  const numberInputRef = useRef<HTMLInputElement>(null);
+  const storageLocationInputRef = useRef<HTMLInputElement>(null);
+
+  const [invalidName] = useState<boolean>(false);
+  const [invalidNumber, setInvalidNumber] = useState<boolean>(false);
+  const [isChecked, setIsChecked] = useState<boolean>(true);
+  const [agencyInfo, setAgencyInfo] = useState<ApplicantInfo>();
+  const [storageLocationEmpty, setStorageLocationEmpty] = useState<boolean>(false);
+
+  const getSeedlotData = () => {
+    if (seedlot) {
+      axios.get(getUrl(ApiAddresses.SeedlotRetrieveOne).replace(':seedlotnumber', seedlot), getAxiosConfig())
+        .then((response) => {
+          if (response.data.seedlotApplicantInfo.applicant) {
+            setAgencyInfo(response.data.seedlotApplicantInfo.applicant);
+          }
+        })
+        .catch((error) => {
+          // eslint-disable-next-line
+          console.error(`Error: ${error}`);
+        });
+    }
+  };
+
+  const interimStorageData: InterimStorageRegistration = {
+    seedlotNumber: (seedlot ? +seedlot : 0),
+    applicant: {
+      name: (agencyInfo ? agencyInfo.name : mockAgencyOptions[0]),
+      number: (agencyInfo ? agencyInfo.number : '0')
+    },
+    storageInformation: {
+      startDate: '',
+      endDate: '',
+      location: ''
+    },
+    facilityType: 'outside'
+  };
+
+  const [responseBody, setResponseBody] = useState<InterimStorageRegistration>(interimStorageData);
+
+  useEffect(() => {
+    getSeedlotData();
+  });
+
+  const inputChangeHandlerStorage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setResponseBody({
+      ...responseBody,
+      applicant: {
+        ...responseBody.applicant,
+        [name]: value
+      }
+    });
+  };
+
+  const inputChangeHandlerLocation = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
```

Comment:
`inputChangeHandlerStorage` and `inputChangeHandlerLocation` does the same thing, we should combine them

Reply:
all the input handler functions that does `setResponseBody` at the end should be combined into one function